packages/jupyter-ai/src/components/mui-extras/async-icon-button.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useMemo, useState } from 'react';
```

Comment:
The `mui-extras` confused me when I first saw it, because I presumed this was part of the MUI distribution. Naming this subdirectory `ui/` would be less ambiguous.